navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/MapboxNavigationOptions.java

```diff
@@ -0,0 +1,263 @@
+package com.mapbox.services.android.navigation.v5;
+
+import android.location.Location;
+import android.support.annotation.FloatRange;
+
+public class MapboxNavigationOptions {
+
+  private double maxTurnCompletionOffset = NavigationConstants.MAXIMUM_ALLOWED_DEGREE_OFFSET_FOR_TURN_COMPLETION;
+  private double maneuverZoneRadius = NavigationConstants.MANEUVER_ZONE_RADIUS;
+
+  private int mediumAlertInterval = NavigationConstants.MEDIUM_ALERT_INTERVAL;
+  private int highAlertInterval = NavigationConstants.HIGH_ALERT_INTERVAL;
+
+  private double minimumMediumAlertDistance = NavigationConstants.MINIMUM_DISTANCE_FOR_MEDIUM_ALERT;
+  private double minimumHighAlertDistance = NavigationConstants.MINIMUM_DISTANCE_FOR_HIGH_ALERT;
+
+  private double maximumDistanceOffRoute = NavigationConstants.MAXIMUM_DISTANCE_BEFORE_OFF_ROUTE;
+  private double deadReckoningTimeInterval = NavigationConstants.DEAD_RECKONING_TIME_INTERVAL;
+  private double maxManipulatedCourseAngle = NavigationConstants.MAX_MANIPULATED_COURSE_ANGLE;
+
+  /**
+   * Creates a new MapboxNavigationOptions object.
+   *
+   * @since 0.2.0
+   */
+  public MapboxNavigationOptions() {
+  }
+
+  /**
+   * Set the threshold user must be within to count as completing a
+   * {@link com.mapbox.services.api.directions.v5.models.LegStep}. When checking if the users completed the step, the
+   * users {@link Location#getBearing()} and the
+   * {@link com.mapbox.services.api.directions.v5.models.StepManeuver#bearingAfter} are compared. If this number is
+   * within the {@code setMaxTurnCompletionOffset} the user has completed the step. This is one of the two heuristics
+   * used to know when a user completes a step, the other being {@link NavigationConstants#MANEUVER_ZONE_RADIUS}.
+   *
+   * @param maxTurnCompletionOffset degree double value between 0 and 359 representing the threshold used to consider if
+   *                                the user has completed the step maneuver.
+   * @return This.
+   * @since 0.2.0
+   */
+  public MapboxNavigationOptions setMaxTurnCompletionOffset(
+    @FloatRange(from = 0, to = 359) double maxTurnCompletionOffset) {
+    this.maxTurnCompletionOffset = maxTurnCompletionOffset;
+    return this;
```

Comment:
discussed internally, since no logic is happening inside `MapboxNavigationOptions` it is a builder in itself and has no need for the same builder pattern inside Mapbox Java.